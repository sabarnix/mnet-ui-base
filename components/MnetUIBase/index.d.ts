import * as React from "react";
import { ThemeType } from "../../themes";
import { BackgroundType } from '../../utils';

export interface MnetUIBaseProps {
  background?: BackgroundType;
  cssVars?: boolean;
  dir?: "rtl";
  full?: boolean;
  plain?: boolean;
  theme?: ThemeType;
  themeMode?: "dark" | "light";
  userAgent?: string;
}

declare const MnetUIBase: React.ComponentClass<MnetUIBaseProps & JSX.IntrinsicElements['div']>;

export { MnetUIBase };
