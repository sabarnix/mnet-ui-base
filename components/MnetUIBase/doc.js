"use strict";

exports.__esModule = true;
exports.themeDoc = exports.doc = void 0;

var _reactDesc = require("react-desc");

var _utils = require("../../utils");

var doc = function doc(MnetUIBase) {
  var DocumentedMnetUIBase = (0, _reactDesc.describe)(MnetUIBase).availableAt((0, _utils.getAvailableAtBadge)('MnetUIBase')).description('The top level MnetUIBase container.').usage("import { MnetUIBase } from 'mnet-ui-base';\n<MnetUIBase>...</MnetUIBase>").intrinsicElement('div');
  DocumentedMnetUIBase.propTypes = {
    background: _utils.backgroundDoc,
    dir: _reactDesc.PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: _reactDesc.PropTypes.bool.description('Whether to take the whole viewport.').defaultValue(false),
    plain: _reactDesc.PropTypes.bool.description("Whether or not MnetUIBase should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: _reactDesc.PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: _reactDesc.PropTypes.object.description('Custom styles for MnetUIBase app component.'),
    themeMode: _reactDesc.PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: _reactDesc.PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint.")
  };
  return DocumentedMnetUIBase;
};

exports.doc = doc;
var themeDoc = {
  'mnet.extend': {
    description: 'Any additional style for MnetUIBase.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};
exports.themeDoc = themeDoc;