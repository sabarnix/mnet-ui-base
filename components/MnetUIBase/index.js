"use strict";

exports.__esModule = true;
exports.MnetUIBase = void 0;

var _MnetUIBase = require("./MnetUIBase");

exports.MnetUIBase = _MnetUIBase.MnetUIBase;