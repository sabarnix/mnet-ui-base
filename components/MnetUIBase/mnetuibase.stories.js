"use strict";

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _mnetUiBase = require("mnet-ui-base");

var _grommetIcons = require("grommet-icons");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return /*#__PURE__*/_react["default"].createElement(_mnetUiBase.MnetUIBase, {
    theme: customTheme
  }, /*#__PURE__*/_react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium"
  }, /*#__PURE__*/_react["default"].createElement(_mnetUiBase.Anchor, {
    icon: /*#__PURE__*/_react["default"].createElement(_grommetIcons.Add, null),
    label: "Add",
    color: "custom"
  })));
};

var Plain = function Plain() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_mnetUiBase.MnetUIBase, {
    plain: true
  }, /*#__PURE__*/_react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium"
  }, /*#__PURE__*/_react["default"].createElement("p", null, "Plain MnetUIBase"))), /*#__PURE__*/_react["default"].createElement(_mnetUiBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium"
  }, /*#__PURE__*/_react["default"].createElement("p", null, "Not plain MnetUIBase"))));
};

var MnetUIBaseVars = function MnetUIBaseVars() {
  return /*#__PURE__*/_react["default"].createElement(_mnetUiBase.MnetUIBase, {
    theme: _mnetUiBase.mnet,
    cssVars: true
  }, /*#__PURE__*/_react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium",
    background: "var(--accent-2)",
    gap: "medium"
  }, /*#__PURE__*/_react["default"].createElement(_mnetUiBase.Box, null, "Checkout MnetUIBase variables, you can find them in the StyledMnetUIBase DOM."), /*#__PURE__*/_react["default"].createElement(_mnetUiBase.Box, {
    "with": true
  }, "For example, the background color in this Box is using var(--accent-2)")));
};

(0, _react2.storiesOf)('MnetUIBase', module).add('Plain', function () {
  return /*#__PURE__*/_react["default"].createElement(Plain, null);
}).add('Theme', function () {
  return /*#__PURE__*/_react["default"].createElement(Themed, null);
}).add('Vars', function () {
  return /*#__PURE__*/_react["default"].createElement(MnetUIBaseVars, null);
});