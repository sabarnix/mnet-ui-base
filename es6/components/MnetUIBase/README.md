## MnetUIBase
The top level MnetUIBase container.

[![](https://cdn-images-1.medium.com/fit/c/120/120/1*TD1P0HtIH9zF0UEH28zYtw.png)](https://storybook.grommet.io/?selectedKind=MnetUIBase&full=0&addons=0&stories=1&panelRight=0) [![](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/grommet/grommet-sandbox?initialpath=mnetuibase&module=%2Fsrc%2FMnetUIBase.js)
## Usage

```javascript
import { MnetUIBase } from 'mnet-ui-base';
<MnetUIBase>...</MnetUIBase>
```

## Properties

**background**

Either a color 
identifier to use for the background color. For example: 'neutral-1'. Or, a 
'url()' for an image. Dark is not needed if color is provided.

```
string
{
  color: 
    string
    {
      dark: string,
      light: string
    },
  dark: 
    boolean
    string,
  image: string,
  position: string,
  opacity: 
    string
    boolean
    number
    weak
    medium
    strong,
  repeat: 
    no-repeat
    repeat
    string,
  size: 
    cover
    contain
    string,
  light: string
}
```

**dir**

Layout direction for right to left contexts

```
rtl
```

**full**

Whether to take the whole viewport.

```
boolean
```

**plain**

Whether or not MnetUIBase should apply a global font-family, font-size,
        and line-height.

```
boolean
```

**cssVars**

Whether to expose the css variables.

```
boolean
```

**theme**

Custom styles for MnetUIBase app component.

```
object
```

**themeMode**

Dark vs. light theme variation. Default is unspecified and left to
      theme.

```
dark
light
```

**userAgent**

User agent used to detect the device width for setting the initial
      breakpoint.

```
string
```
  
## Intrinsic element

```
div
```
## Theme
  
**mnet.extend**

Any additional style for MnetUIBase. Expects `string | (props) => {}`.

Defaults to

```
undefined
```

**global.font.face**

Custom font face declaration Expects `string | (props) => {}`.

Defaults to

```
undefined
```
