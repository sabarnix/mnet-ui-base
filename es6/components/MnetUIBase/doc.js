import { describe, PropTypes } from 'react-desc';
import { backgroundDoc, getAvailableAtBadge } from '../../utils';
export var doc = function doc(MnetUIBase) {
  var DocumentedMnetUIBase = describe(MnetUIBase).availableAt(getAvailableAtBadge('MnetUIBase')).description('The top level MnetUIBase container.').usage("import { MnetUIBase } from 'mnet-ui-base';\n<MnetUIBase>...</MnetUIBase>").intrinsicElement('div');
  DocumentedMnetUIBase.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: PropTypes.bool.description('Whether to take the whole viewport.').defaultValue(false),
    plain: PropTypes.bool.description("Whether or not MnetUIBase should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: PropTypes.object.description('Custom styles for MnetUIBase app component.'),
    themeMode: PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint.")
  };
  return DocumentedMnetUIBase;
};
export var themeDoc = {
  'mnet.extend': {
    description: 'Any additional style for MnetUIBase.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};