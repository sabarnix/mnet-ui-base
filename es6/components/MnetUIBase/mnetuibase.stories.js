import React from 'react';
import { storiesOf } from '@storybook/react';
import { mnet, MnetUIBase, Anchor, Box } from 'mnet-ui-base';
import { Add } from "grommet-icons/es6/icons/Add";
var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: customTheme
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement(Anchor, {
    icon: /*#__PURE__*/React.createElement(Add, null),
    label: "Add",
    color: "custom"
  })));
};

var Plain = function Plain() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(MnetUIBase, {
    plain: true
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement("p", null, "Plain MnetUIBase"))), /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement("p", null, "Not plain MnetUIBase"))));
};

var MnetUIBaseVars = function MnetUIBaseVars() {
  return /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: mnet,
    cssVars: true
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium",
    background: "var(--accent-2)",
    gap: "medium"
  }, /*#__PURE__*/React.createElement(Box, null, "Checkout MnetUIBase variables, you can find them in the StyledMnetUIBase DOM."), /*#__PURE__*/React.createElement(Box, {
    "with": true
  }, "For example, the background color in this Box is using var(--accent-2)")));
};

storiesOf('MnetUIBase', module).add('Plain', function () {
  return /*#__PURE__*/React.createElement(Plain, null);
}).add('Theme', function () {
  return /*#__PURE__*/React.createElement(Themed, null);
}).add('Vars', function () {
  return /*#__PURE__*/React.createElement(MnetUIBaseVars, null);
});