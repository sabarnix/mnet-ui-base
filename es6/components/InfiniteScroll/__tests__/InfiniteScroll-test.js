import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { MnetUIBase } from '../../MnetUIBase';
import { InfiniteScroll } from '..';
describe('InfiniteScroll', function () {
  var items = [];

  while (items.length < 4) {
    items.push(items.length);
  }

  test('basic', function () {
    var _render = render( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(InfiniteScroll, null), /*#__PURE__*/React.createElement(InfiniteScroll, {
      items: items
    }, function (item, index, ref) {
      return /*#__PURE__*/React.createElement("div", {
        ref: ref,
        key: index
      }, item);
    }), /*#__PURE__*/React.createElement(InfiniteScroll, {
      items: items
    }, function (item, index) {
      return /*#__PURE__*/React.createElement("div", {
        key: index
      }, item);
    }))),
        container = _render.container;

    expect(container.firstChild).toMatchSnapshot();
  });
  test('step', function () {
    var _render2 = render( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(InfiniteScroll, {
      items: items,
      step: 2
    }, function (item, index) {
      return /*#__PURE__*/React.createElement("div", {
        key: index
      }, item);
    }))),
        container = _render2.container;

    expect(container.firstChild).toMatchSnapshot();
  });
  test('show', function () {
    var _render3 = render( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(InfiniteScroll, {
      items: items,
      step: 2,
      show: 3
    }, function (item, index) {
      return /*#__PURE__*/React.createElement("div", {
        key: index
      }, item);
    }))),
        container = _render3.container;

    expect(container.firstChild).toMatchSnapshot();
  });
  test('renderMarker', function () {
    var _render4 = render( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(InfiniteScroll, {
      items: items,
      step: 2,
      renderMarker: function renderMarker(m) {
        return /*#__PURE__*/React.createElement("div", null, m);
      }
    }, function (item, index) {
      return /*#__PURE__*/React.createElement("div", {
        key: index
      }, item);
    }))),
        container = _render4.container;

    expect(container.firstChild).toMatchSnapshot();
  });
  test('replace', function () {
    var _render5 = render( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(InfiniteScroll, {
      items: items,
      step: 2,
      replace: true
    }, function (item, index) {
      return /*#__PURE__*/React.createElement("div", {
        key: index
      }, item);
    }))),
        container = _render5.container;

    expect(container.firstChild).toMatchSnapshot();
  });
});