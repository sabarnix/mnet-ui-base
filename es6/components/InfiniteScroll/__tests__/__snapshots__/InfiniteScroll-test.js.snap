// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteScroll basic 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
</div>
`;

exports[`InfiniteScroll renderMarker 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    <div
      class="c1"
    />
  </div>
</div>
`;

exports[`InfiniteScroll replace 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div
    class="c1"
  />
</div>
`;

exports[`InfiniteScroll show 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div
    class="c1"
  />
</div>
`;

exports[`InfiniteScroll step 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 0px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div>
    0
  </div>
  <div>
    1
  </div>
  <div
    class="c1"
  />
</div>
`;
